content/doc/developer/extensibility/action-for-all-projects.adoc

```diff
@@ -0,0 +1,144 @@
+---
+title: How to show a menu item on all projects
+layout: developerguide
+---
+
+This guide will guide you through showing a menu item with associated URL and view on all projects' side panels, without requiring configuration in the job.
+
+This could be used for example to show information about the job, such as statistics, without needing additional (persistently stored) data.
+
+In this example, we'll add a link called _Statistics_ that will link to a page that shows some information about the project.
+
+## Create the action
+
+The action will be fairly basic: It will expect a reference to an +jenkinsdoc:Project[]+ as constructor argument, and has getters returning the number of build steps and post-build steps this project has.
+
+[listing]
+----
+package org.jenkinsci.plugins.sample;
+
+import hudson.model.Action;
+import hudson.model.Project;
+
+public class SampleAction implements Action {
+
+    private Project project;
+
+    public SampleAction(Project project) {
+        this.project = project;
+    }
+
+    public int getBuildStepsCount() {
+        return project.getBuilders().size();
+    }
+
+    public int getPostBuildStepsCount() {
+        return project.getPublishersList().size();
+    }
+
+    @Override
+    public String getIconFileName() {
+        return "clipboard.png";
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Project Statistics";
+    }
+
+    @Override
+    public String getUrlName() {
+        return "stats";
+    }
+}
+----
+
+Similarly, the `index.jelly` view we create for it in the resource directory corresponding to this class -- +src/main/resources/org/jenkinsci/plugins/sample/SampleAction/+ -- is very basic, just showing the information from the getters:
+
+[listing]
+----
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:l="/lib/layout">
+    <l:layout title="Project Statistics">
+        <l:main-panel>
+            <h1>
+                Project Statistics
```

Comment:
That's really out of scope for a how-to guide. I'm already stretching it to include permissions, and the only reason is that this is closely related (and permissions inform how to implement this, as discussed).